Sources/AirbnbSwiftFormatTool/SwiftSettings.swift

```diff
@@ -0,0 +1,107 @@
+// Created by Cal Stephens on 9/5/23.
+// Copyright © 2023 Airbnb Inc. All rights reserved.
+
+import ArgumentParser
+import Foundation
+import RegexBuilder
+
+// MARK: - SwiftSettings
+
+enum SwiftSettings {
+  /// Default Swift compiler flags recommended by the Airbnb Swift Style Guide.
+  ///
+  /// If targeting `swift-tools-version: 5.8` or later, you can include this
+  /// extension in your `Package.swift` and use it in your targets like
+  /// `swiftSettings: .airbnbDefault()`.
+  ///
+  /// This tool will also keep the `airbnbDefault` settings in your Package.swift
+  /// up to date automatically, if it detects an `extension [SwiftSetting]` with
+  /// the `airbnbDefault` settings already defined.
+  static let airbnbDefaultSwiftSettings = """
+    extension [SwiftSetting] {
+      /// Default Swift compiler flags recommended by the Airbnb Swift Style Guide.
+      /// Do not modify: updated automatically by Airbnb Swift Format Tool.
+      static func airbnbDefault() -> [SwiftSetting] {
+        [
+          .enableUpcomingFeature("BareSlashRegexLiterals"),
+          .enableUpcomingFeature("ConciseMagicFile"),
+          .enableUpcomingFeature("ImplicitOpenExistentials"),
+        ]
+      }
+    }
+    """
+}
+
+extension SwiftSettings {
+
+  // MARK: Internal
+
+  /// Finds all of the `Package.swift` package manifest files in the given package directory,
+  /// and updates the `[SwiftSetting].airbnbDefault()` declaration if present.
+  @available(macOS 13.0, *)
+  static func updatePackageManifests(in packageDirectory: URL, lintOnly: Bool, verbose: Bool) throws {
+    for packageManifestURL in packageManifestURLs(in: packageDirectory) {
+      var contents = try String(contentsOf: packageManifestURL)
+      let originalContent = contents
+
+      let extensionRegex = Regex {
+        "extension [SwiftSetting] {"
+        OneOrMore { .any }
+        "static func airbnbDefault"
+        OneOrMore { .any }
```

Comment:
I wanted to use `/.*/` here, to demonstrate that we're able to use regex literals now in this module. This causes SwiftLint to fail with a parsing error though. In a follow up I can look in to updating to a SwiftLint version that supports this syntax.

Reply:
Fixed with https://github.com/airbnb/swift/pull/236/commits/7ddea2564f901b08e50b24bf5821c2cc0ec281b5 and https://github.com/airbnb/swift/pull/236/commits/e080364df3cd28d37a58812829de3fd8e9d63b90